### DISCUSSION TITLE
Can't save in VSCode

### DISCUSSION THREAD

**Discussion by Stevepurpose** [pieces-app/support#244] [Q&A]
Hello devs. For some reason I cannot save code using the pieces VSCode extension. Also when I ask copilot for a question, I get an almost there... response tht hangs forever.

Do I need to use internet to do these??. While installing, the various tools using snap on Linux Ubuntu, I used the `sudo snap connect pieces-os:process-control :process-control`. I learned it enables offline mode  functionality and on-device machine learning. Please what could be wrong?.

**Comment by Stevepurpose** [comment]
Ok. The VSCode problem has been sorted. Looks like all I needed was to restart my pc.

**Reply by caleb-at-pieces** [reply]
@Stevepurpose thank you for reaching out about this, there is an issue in the current version of the VS Code extension where if you try to save a snippet from a file that has an extension that is unknown to our systems, it will silently fail.

This should be fixed in the next release of the VS Code extension, saving snippets from file extensions that are recognized should work fine.

**Reply by Stevepurpose** [reply]
Ok. Thanks a lot Caleb
